src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTag.cs

```diff
@@ -15,30 +18,50 @@ namespace Microsoft.CodeAnalysis.Editor.InlineParameterNameHints
     /// </summary>
     internal class InlineParameterNameHintsTag : IntraTextAdornmentTag
     {
+        public const string TagId = "inline parameter name hints";
+
         /// <summary>
         /// Creates the UIElement on call
         /// Uses PositionAffinity.Successor because we want the tag to be associated with the following character
         /// </summary>
         /// <param name="text">The name of the parameter associated with the argument</param>
-        public InlineParameterNameHintsTag(string text)
-            : base(CreateElement(text), removalCallback: null, PositionAffinity.Successor)
+        public InlineParameterNameHintsTag(string text, double lineHeight, TextFormattingRunProperties format)
+            : base(CreateElement(text, lineHeight, format), removalCallback: null, PositionAffinity.Successor)
         {
         }
 
-        private static UIElement CreateElement(string text)
+        private static UIElement CreateElement(string text, double lineHeight, TextFormattingRunProperties format)
         {
-            // Constructs the hint block which gets assigned parameter name, a normal fontstyle, and sets the padding 
-            // space around the block to 0
+            // Constructs the hint block which gets assigned parameter name and fontstyles according to the options
+            // page. Calculates a font size 1/4 smaller than the font size of the rest of the editor
             var block = new TextBlock
             {
-                Text = text + ": ",
+                Text = text + ":",
+                VerticalAlignment = VerticalAlignment.Center,
                 FontStyle = FontStyles.Normal,
+                FontFamily = format.Typeface.FontFamily,
+                FontSize = format.FontRenderingEmSize - (0.25 * format.FontHintingEmSize),
```

Comment:
Do we need to be setting both the height of the border and the height of the text? I would have imagined setting one was sufficient and the other would just resize accordingly.

Reply:
I remember reading somewhere that, when encased in a border, you need to have the border set the height in order for the sizing of the text to change relative to that height.